app/src/main/java/com/duckduckgo/app/browser/logindetection/NavigationAwareLoginDetector.kt

```diff
@@ -91,15 +91,20 @@ class NextPageLoginDetection @Inject constructor() : NavigationAwareLoginDetecto
         }
     }
 
-    @Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
     private fun detectLogin(forwardedToUrl: String) {
         val loginAttemptEvent = loginAttempt ?: return
         val loginURI = Uri.parse(loginAttemptEvent.url).takeUnless { it.host.isNullOrBlank() } ?: return
         val forwardedToUri = Uri.parse(forwardedToUrl).takeUnless { it.host.isNullOrBlank() } ?: return
 
         Timber.i("LoginDetectionDelegate ${loginAttemptEvent.url} vs $forwardedToUrl")
         if (loginURI.host != forwardedToUri.host || loginURI.path != forwardedToUri.path) {
-            loginEventLiveData.value = LoginDetected(loginURI.host, forwardedToUri.host)
+            val loginUriHost = loginURI.host
+            val forwardedToUriHost = forwardedToUri.host
+
+            if (loginUriHost != null && forwardedToUriHost != null) {
+                loginEventLiveData.value = LoginDetected(loginUriHost, forwardedToUriHost)
+            }
+
```

Comment:
@cmonfortep, as discussed, you probably want to make a few small changes here?

Reply:
Just pushed a change related to this part 🚀